models/classes/Security/DataAccess/Repository/ToolKeyChainRepository.php

```diff
@@ -38,8 +39,9 @@ public function save(KeyChain $keyChain): void
 
     public function findAll(KeyChainQuery $query): KeyChainCollection
     {
-        $ltiProvider = $this->getLtiProviderService()
-            ->searchById($query->getIdentifier());
+        if (!$ltiProvider = $this->getLtiProviderService()->searchById($query->getIdentifier())) {
```

Comment:
This is a code smell, we should not have variable attribution inside a condition.